scipy/stats/_continuous_distns.py

```diff
@@ -2900,6 +2900,186 @@ def _entropy(self, c):
+            (\hat{\alpha} \sqrt{1 + x^2})}
+            {(\sqrt{1 + x^2})^{1/2 - \lambda}}
+
+    for :math:`x, \lambda \in \mathbb{R}`,
+    :math:`|\hat{\beta}| \lt \hat{\alpha}` if :math:`\lambda \ge 0`,
+    :math:`|\hat{\beta}| \lt \hat{\alpha}` if :math:`\lambda = 0`,
+    :math:`|\hat{\beta}| \leq \hat{\alpha}` if :math:`\lambda \lt 0`.
+    :math:`K_{\lambda}(.)` denotes the modified Bessel function of the second
+    kind and order :math:`\lambda` (`scipy.special.kn`)
+
+    `genhyperbolic` takes ``p`` as a tail parameter for :math:`\lambda`,
+    ``a`` as a shape parameter for :math:`\hat{\alpha}`,
+    ``b`` as a skewness parameter for :math:`\hat{\beta}`,
+
+    Moments are based on [3]_.
+
+    %(after_notes)s
+
+    The original parameterization of the Generalized Hyperbolic Distribution
+    is found in [1]_ as follows
+
+    .. math::
+
+        f(x, \lambda, \alpha, \beta, \delta, \mu) =
+           \frac{(\gamma/\delta)^\lambda}{\sqrt{2\pi}K_\lambda(\delta \gamma)}
+           e^{\beta (x - \mu)} \times \frac{K_{\lambda - 1/2}
+           (\alpha \sqrt{\delta^2 + (x - \mu)^2})}
+           {(\sqrt{\delta^2 + (x - \mu)^2} / \alpha)^{1/2 - \lambda}}
+
+    for :math:`x \in ( - \infty; \infty)`,
+    :math:`\gamma := \sqrt{\alpha^2 - \beta^2}`,
+    :math:`\lambda, \mu \in \mathbb{R}`,
+    :math:`|\beta| \lt \alpha` if :math:`\lambda \ge 0`,
+    :math:`|\beta| \lt \alpha` if :math:`\lambda = 0`,
+    :math:`|\beta| \leq \alpha ` if :math:`\lambda \lt 0`.
+    and :math:`\delta \in \mathbb{R}_{>0}`.
+
+    The location-scale-based parameterization implemented in
+    scipy is based on [2]_, where :math:`\hat{\alpha} = \alpha\delta`
+    and :math:`\hat{\beta} = \beta\delta`
+
+    References
+    ----------
+    .. [1] O. Barndorff-Nielsen, "Hyperbolic Distributions and Distributions
+       on Hyperbolae", Scandinavian Journal of Statistics, Vol. 5(3),
+       pp. 151-157, 1978. https://www.jstor.org/stable/4615705
+
+    .. [2] Eberlein E., Prause K. (2002) The Generalized Hyperbolic Model:
+        Financial Derivatives and Risk Measures. In: Geman H., Madan D.,
+        Pliska S.R., Vorst T. (eds) Mathematical Finance - Bachelier
+        Congress 2000. Springer Finance. Springer, Berlin, Heidelberg.
+        https://doi.org/10.1007/978-3-662-12429-1_12
+
+    .. [3] Scott, David J, Würtz, Diethelm, Dong, Christine and Tran,
+       Thanh Tam, (2009), Moments of the generalized hyperbolic
+       distribution, MPRA Paper, University Library of Munich, Germany,
+       https://EconPapers.repec.org/RePEc:pra:mprapa:19081.
+
+    %(example)s
+
+    """
+
+    def _argcheck(self, p, a, b):
+
+        @np.vectorize
+        def _argcheck_single(p, a, b):
+            if p > 0:
+                return np.absolute(b) < a
+            elif p == 0:
+                return np.absolute(b) < a
+            else:
+                return np.absolute(b) <= a
+
+        return _argcheck_single(p, a, b)
```

Comment:
```suggestion
return np.logical_and(np.abs(b) < a, p >= 0) | np.logical_and(np.abs(b) <= a, p < 0)
```
I think that's close, anyway.